projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -1066,28 +1066,29 @@ export function insertCommentAboveIdentifier(
   comment: string,
   identifierType = ts.SyntaxKind.Identifier
 ): Change[] {
-  const classNode = getSourceNodes(source).find(
-    (node) => node.kind === ts.SyntaxKind.ClassDeclaration
-  );
-  if (!classNode) {
-    return [new NoopChange()];
-  }
+  const changes: InsertChange[] = [];
 
-  const identifierNodes = findNodes(classNode, identifierType).filter(
-    (node) => node.getText() === identifierName
-  );
+  getSourceNodes(source).forEach((node) => {
+    if (node.kind !== ts.SyntaxKind.ClassDeclaration) {
+      return;
+    }
 
-  const changes: InsertChange[] = [];
-  identifierNodes.forEach((n) =>
-    changes.push(
-      new InsertChange(
-        sourcePath,
-        getLineStartFromTSFile(source, n.getStart()),
-        `${comment}`
+    const identifierNodes = findNodes(node, identifierType).filter(
+      (node) => node.getText() === identifierName
+    );
+
+    identifierNodes.forEach((n) =>
+      changes.push(
+        new InsertChange(
+          sourcePath,
+          getLineStartFromTSFile(source, n.getStart()),
+          `${comment}`
+        )
       )
-    )
-  );
-  return changes;
+    );
+  });
+
+  return changes.length ? changes : [new NoopChange()];
```

Comment:
You can just return `changes` here

Reply:
In previous implementation we returned `[new NoopChange()]` when there was no change. Do you think we can safely drop it?

Reply:
Yes, for sure. It might even be a bit more optimal, as we check later on for the length of the changes array.